src/renderers/webgl/WebGLProgram.js

```diff
@@ -810,6 +822,12 @@ function WebGLProgram( renderer, cacheKey, parameters, bindingStates ) {
 
 		if ( cachedUniforms === undefined ) {
 
+			if ( firstUse ) {
```

Comment:
Is this change here required? Or is it possible to always call `onFirstUse()` when `cachedUniforms` is `undefined`?

Reply:
It was done this way because `onFirstUse()` could be called from either the `getUniforms()` or `getAttributes()` and we didn't want to call it twice. I've updated it so that `onFirstUse()` populates both the uniforms and attributes, which should be fine because the chances you will need to use only uniforms or attributes but not both is pretty literally 0 and the primary cost is waiting for the shader compile to finish, not either of those individual queries.